mono/mini/intrinsics.c

```diff
@@ -736,11 +736,11 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 		} else
 			return NULL;
 	} else if (cmethod->klass == mono_defaults.array_class) {
-		if (strcmp (cmethod->name, "GetGenericValueImpl") == 0 && fsig->param_count + fsig->hasthis == 3 && !cfg->gsharedvt)
+		if (fsig->param_count + fsig->hasthis == 3 && !cfg->gsharedvt && strcmp (cmethod->name, "GetGenericValueImpl") == 0)
 			return emit_array_generic_access (cfg, fsig, args, FALSE);
-		else if (strcmp (cmethod->name, "SetGenericValueImpl") == 0 && fsig->param_count + fsig->hasthis == 3 && !cfg->gsharedvt)
+		else if (fsig->param_count + fsig->hasthis == 3 && !cfg->gsharedvt && strcmp (cmethod->name, "SetGenericValueImpl") == 0)
 			return emit_array_generic_access (cfg, fsig, args, TRUE);
-		else if (!strcmp (cmethod->name, "GetRawSzArrayData")) {
+		else if (!strcmp (cmethod->name, "GetRawSzArrayData") || !strcmp (cmethod->name, "GetRawArrayData")) {
```

Comment:
Reviewers please do note that I have opportunistically added an intrinsic here.
This is the "extended" part of the PR description.
I just happened to notice, that its current implementation is identical to the other function.
I do not know:
 1. If it is called often, is performance critical, what difference this made.
 2. If it is a temporary coincidence or long term architectural thing. It does seem just like a rename of the other, so long term architectural.

However the intrinsic notation (or even a comment) should assist in case I am wrong.
i.e. anyone changing the implementation needs to heed the comment/attribute and either leave it alone, or update the JIT, or, worst case, stop it from being an intrinsic and just use the C#, i.e. undo this change as needed.